src/Mvc/Mvc.ViewFeatures/src/ViewComponentResultExecutor.cs

```diff
@@ -149,8 +153,8 @@ public virtual async Task ExecuteAsync(ActionContext context, ViewComponentResul
                 }
                 else
                 {
-                    using var bufferingStream = new FileBufferingWriteStream();
-                    using (var intermediateWriter = _writerFactory.CreateWriter(bufferingStream, resolvedContentTypeEncoding))
+                    using var bufferingStream = _bufferFactory.CreateWriteStream();
```

Comment:
Could we instead of resolving in Mvc `IWebHostEnvironment`, register and resolve a Factory for the BufferedStreams? That way we are not creating the concrete types here, and resolving a factory allows us to register another one, for testing for example. 

As a test I did it in `ViewComponentResultExecutor`, registering the `IFileBufferingStreamFactory`, but probably with a better naming like`IBufferedStreamFactory` would be better. 

What do you think?

Reply:
That's right back where we were before we moved it to IWebHostEnvironment. That service needs to be registered by someone. I'll discuss this with @davidfowl.